File: Examen1/2026_1/TestSoft/TestSoftNegocio/src/main/java/pe/edu/pucp/testsoft/negocio/pregunta/boimpl/PreguntaBOImpl.java
package pe.edu.pucp.testsoft.negocio.pregunta.boimpl;

import pe.edu.pucp.testsoft.model.pregunta.Pregunta;
import pe.edu.pucp.testsoft.negocio.pregunta.bo.PreguntaBO;
import pe.edu.pucp.testsoft.persistance.pregunta.dao.PreguntaDAO;
import pe.edu.pucp.testsoft.persistance.pregunta.impl.PreguntaImpl;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class PreguntaBOImpl implements PreguntaBO {

    private PreguntaDAO preguntaDAO = new PreguntaImpl();

    @Override
    public int insertar(Pregunta objeto) throws Exception {
        return preguntaDAO.insertar(objeto);
    }

    @Override
    public int modificar(Pregunta objeto) throws Exception {
        return preguntaDAO.modificar(objeto);
    }

    @Override
    public int eliminar(int id) throws Exception {
        return preguntaDAO.eliminar(id);
    }

    @Override
    public Pregunta buscarPorId(int id) throws Exception {
        return preguntaDAO.buscarPorId(id);
    }

    @Override
    public List<Pregunta> leerTodos() {
        return preguntaDAO.listarTodos();
    }

    @Override
    public List<Pregunta> seleccionarPreguntasAleatorias() {
        List<Pregunta> todas = preguntaDAO.listarTodos();
        if (todas == null || todas.isEmpty()) return new ArrayList<>();

        // Mezclar y tomar las primeras 10 (o menos si hay menos)
        Collections.shuffle(todas);
        int cantidad = Math.min(10, todas.size());
        return new ArrayList<>(todas.subList(0, cantidad));
    }
}

File: Examen1/2026_1/PlantillaC#/TestSoft/TestSoftBusiness/Preguntas/BO/PreguntaBOImpl.cs
using TestSoftModel.Pregunta;
using TestSoftBusiness.Preguntas.BOI;
using TestSoftPersistance.Preguntas.DAO;
using TestSoftPersistance.Preguntas.Impl;

namespace TestSoftBusiness.Preguntas.BO
{
    public class PreguntaBOImpl : IPreguntaBO
    {
        private readonly PreguntaDAO preguntaDAO = new PreguntaImpl();

        public int Insertar(Pregunta objeto) => preguntaDAO.Insertar(objeto);
        public int Modificar(Pregunta objeto) => preguntaDAO.Modificar(objeto);
        public int Eliminar(int id) => preguntaDAO.Eliminar(id);
        public Pregunta BuscarPorId(int id) => preguntaDAO.BuscarPorId(id);
        public List<Pregunta> ListarTodos() => preguntaDAO.ListarTodos();

        // Selecciona 10 preguntas aleatorias (o menos si hay menos)
        public List<Pregunta> SeleccionarPreguntasAleatorias()
        {
            List<Pregunta> todas = preguntaDAO.ListarTodos();
            if (todas == null || todas.Count == 0) return new List<Pregunta>();

            Random rng = new Random();
            List<Pregunta> mezcladas = todas.OrderBy(x => rng.Next()).ToList();
            int cantidad = Math.Min(10, mezcladas.Count);
            return mezcladas.Take(cantidad).ToList();
        }
    }
}
